query-engine/connectors/sql-query-connector/src/join_utils.rs

```diff
@@ -226,24 +230,16 @@ pub fn compute_one2m_join(base_model: &ModelRef, rf: &RelationFieldRef, join_pre
         )
     };
 
-    // `rf` is always the relation field on the left model in the join (parent).
-    let left_table_alias = if rf.model().name != base_model.name {
-        Some(format!("{}_{}", join_prefix, &rf.model().name))
-    } else {
-        None
-    };
-
     let right_table_alias = format!("{}_{}", join_prefix, &rf.related_model().name);
 
     let related_model = rf.related_model();
     let pairs = left_fields.into_iter().zip(right_fields.into_iter());
 
     let on_conditions: Vec<Expression> = pairs
         .map(|(a, b)| {
-            let a_col = if let Some(alias) = left_table_alias.clone() {
-                Column::from((alias, a.db_name().to_owned()))
-            } else {
-                a.as_column()
+            let a_col = match previous_join {
+                Some(prev_join) => Column::from((prev_join.alias.to_owned(), a.db_name().to_owned())),
+                None => a.as_column(),
             };
```

Comment:
This ensures that if we do something like: `{ orderBy: { rel: { self_rel: { id: asc } } } }`, the second join relies on the first one.

We used to make sure joins relied on each other slightly differently. This should be more bullet-proof.